lib/iris/tests/unit/io/test_expand_filespecs.py

```diff
@@ -96,6 +96,12 @@ def test_files_and_none(self):
 
         self.assertMultiLineEqual(str(err.exception), expected)
 
+    def test_false_bool(self):
+        tempdir = self.tmpdir
+        msg = os.path.join(tempdir, "no_exist.txt")
+        result = iio.expand_filespecs([msg], False)[0]
+        self.assertEqual(result, msg)
+
```

Comment:
I think there should be a testcase here that confirms tilde expansion.
I think you can do that, probably, by patching the value of `os.environ['HOME']` to ensure a consistent answer.